src/models/RecentlyListened.js

```diff
@@ -0,0 +1,37 @@
+const mongoose = require('mongoose');
+
+const recentlyListenedSchema = new mongoose.Schema({
+  user: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'User',
+    required: true,
+  },
+  artistName: {
+    type: String,
+    required: true,
+    //AKOS: should i delete this field because we have no artist model and we can't  reference it here  ?
```

Comment:
We could leave it like this, and that means when creating a `RecentlyListened` object we'd need to do something like: `await RecentlyListened.create({user, album, artistName: album.artistName})`.

But then that leaves the possibility for bugs where someone accidentally passes in a different artist that doesn't match the album name.

I think a better way would be to have `artistName` be a [virtual field](https://mongoosejs.com/docs/tutorials/virtuals.html) that just pulls the value from the `album`:

```js
recentlyListenedSchema.virtual('artistName', { ref: 'Album', localField: 'album', foreignField: 'artistName', justOne: true });

// below is in node console:

> let x = await RecentlyListened.findOne({})
undefined
> x.artistName
'Luke Bryan'
> x
{
  _id: new ObjectId("665f69c1150dc898bdd44267"),
  user: new ObjectId("64d6a966a15d2e18ab96a263"),
  album: new ObjectId("64d2ae2308a725b72bd5c0db"),
  timeListened: 2024-06-04T19:23:45.664Z,
  __v: 0
}
```

Or we could leave out the artistName altogether, and just make sure that we are always populating it from the album when fetching like this:

```js
// populate the full album
await RecentlyListened.findOne({}).populate('album')
{
  _id: new ObjectId("665f69c1150dc898bdd44267"),
  user: new ObjectId("64d6a966a15d2e18ab96a263"),
  artistName: 'Luke Bryan',
  album: {
    _id: new ObjectId("64d2ae2308a725b72bd5c0db"),
    artistName: 'Luke Bryan',
    albumName: 'Crash My Party...Again',
    price: 80,
    image: 'https://i.scdn.co/image/ab67616d0000b27322df448bba2c2b671d8502c8',
    releaseDate: 2023-08-04T00:00:00.000Z,
    spotifyUrl: 'https://api.spotify.com/v1/albums/3f4MmjZHwYHNHdnodbOHRe',
    averageRating: 3,
    numOfReviews: 4,
    __v: 0,
    createdAt: 2023-08-08T21:05:39.694Z,
    updatedAt: 2023-12-07T13:22:15.191Z
  },
  timeListened: 2024-06-04T19:23:45.664Z,
  __v: 0
}



// or populate just the artist name
await RecentlyListened.findOne({}).populate('album', 'artistName')
{
  _id: new ObjectId("665f69c1150dc898bdd44267"),
  user: new ObjectId("64d6a966a15d2e18ab96a263"),
  artistName: 'Luke Bryan',
  album: {
    _id: new ObjectId("64d2ae2308a725b72bd5c0db"),
    artistName: 'Luke Bryan'
  },
  timeListened: 2024-06-04T19:23:45.664Z,
  __v: 0
}
```